### DISCUSSION TITLE
Understanding about webpack loaders

### DISCUSSION THREAD

**Discussion by Lotes** [webpack/webpack#18108] [Q&A] [answered]
I made a file format "ABC" which I can transpile to Typescript.

Next, I want to load the transpiled Typescript every time I reference the `.abc` file. Until now I thought this can be done with a custom webpack loader. I saw this working for CSS, where the loader adds simple JSON objects with names of all CSS classes.

```ts
import { $UserType } from "./Blog.abc";

const user: $UserType = {
  $type: "UserType",
  id: 1,
  name: 'John Doe',
  email: '',
  createdAt: '',
  updatedAt: '',
};

console.log(JSON.stringify(user, null, 2));
```

But my loader does not work. So I cloned the CSS loader and just replaced the CSS regexp with my ABC  regexp. But nothing happens.

The loader is imported by the following webpack config line. But a profiling told me that it was not used by  webpack...

```js
{ test: /\.abc$/, use: ["ts-loader", "@abc/webpack-loader"]  },
```

My whole setup is in a monorepo.

Is my understanding of loaders correct?
What am I missing?

**Comment by alexander-akait** [comment]
Weird, should work, can you provide a reproducible example? Maybe you already have another loader, so it was run firstly

**Reply by Lotes** [reply]
Thanks for your help :).

[Here](https://github.com/Lotes/ecchi-js) is my project. Under `packages/example` you can find my final setup with the `webpack.config.js`.

In the meanwhile I found out that the file format is touched, when I use a value and not a type. Very confusing. Now the custom file was read, but nothing gets generated into my `example/out` folder.

I will create some documentation now. But to run the failing test: `npm install && npm test`

EDIT: The problem now is that the output of the loader is nowhere persisted or imported... Do you have an idea? Also my console.log are done twice in the result output...

**Reply by Lotes** [reply]
It seems that my transpiler does not work because of something else

**Reply by Lotes** [reply]
The transpiler seems to do something, and ts-loader seems to transpile my source to JS. But for some reason the symbols are not exported, using `npx webpack --mode=production --stats=detailed`:

```
WARNING in ./src/index.ts 10:12-23
export '$Reflection' (imported as '$Reflection') was not found in './Blog.abc' (module has no exports)
```
Is there a way to get the intermediate result between the loaders? Let's see...

**Reply by Lotes** [reply, marked as answer]
The transpiled code had a syntax error, then ts-loader refuses to work correctly.
